src/materialized/http.rs

```diff
@@ -46,28 +47,57 @@ pub fn match_handshake(buf: &[u8]) -> bool {
     METHODS.contains(&buf)
 }
 
-pub async fn handle_connection<A: 'static + AsyncRead + AsyncWrite + Unpin>(
-    a: A,
-    cmd_tx: UnboundedSender<coord::Command>,
+pub struct Server {
+    cmd_tx: Weak<UnboundedSender<coord::Command>>,
     gather_metrics: bool,
```

Comment:
fwiw in the future I think we should just ditch this gather metrics parameter. If it is ever a perf issue we should just compile them out, they haven't ever shown up in any traces AFAIK, though.

Reply:
👍